network/p2p/gossip/mempool.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+
+	xmempool "github.com/ava-labs/avalanchego/vms/txs/mempool"
+)
+
+var ErrDuplicateTx = xmempool.ErrDuplicateTx
+
+type Mempool[T Gossipable] interface {
```

Comment:
Do we need this interface?

Reply:
I think that an interface better define the functionality we want to provide, in contrast to a struct, which is more of a data-type. It's also more aligned with the rest of our codebase. But I'm definitely open to change that if you think it's over-engineering.

Reply:
> I think that an interface better define the functionality we want to provide, in contrast to a struct, which is more of a data-type.

Why? This is not how interfaces are used in Go because interfaces are defined implicitly. Interfaces are generally defined in the package that consumes a type - not defined by the package that produces the type

> It's also more aligned with the rest of our codebase.

This is true, but the reason earlier parts of our codebase were not written with idiomatic Go in mind is because at the time the team did not understand Go style. We should not continue writing non-idiomatic Go for the sake of consistency.

Reply:
dropped interface, exported struct.